contracts/scripts/deploy_all.ts

```diff
@@ -0,0 +1,79 @@
+require("dotenv").config();
+import util from "util";
+import { exec } from "child_process";
+import { writeFile } from "fs";
+import { ethers } from "hardhat";
+import { NetworkConfig } from "../../clients/src";
+import deployAndRunPrecompileCostEstimator from "../shared/helpers/deployAndRunPrecompileCostEstimator";
+import Fixture, { DOMAIN_HEX } from "../shared/helpers/Fixture";
+
+const execAsync = util.promisify(exec);
+const writeFileAsync = util.promisify(writeFile);
```

Comment:
Nitpick: I'd prefer the callback versions get a -Cb prefix and the promisified versions get the unmodified name.

Also for `fs` you can just import from `fs/promises` which is cleaner and the simpler underlying code may produce less weirdness (e.g. error messages).